paramnormal/dist.py

```diff
@@ -904,6 +906,114 @@ def fit(cls, data, **guesses):
         return cls.param_template(R=b*sigma, loc=loc, sigma=sigma)
 
 
+class truncated_normal(BaseDist_Mixin):
+    """
+    Create and fit data to a truncated normal distribution.
+
+    Methods
+    -------
+    fit
+        Use scipy's maximum likelihood estimation methods to estimate
+        the parameters of the data's distribution.
+    from_params
+        Create a new distribution instances from the ``namedtuple``
+        result of the :meth:`~fit` method.
+
+    Parameters
+    ----------
+    lower, upper : float
+        The lower and upper limits of the distribution that serve as its
+        shape parameters.
+    mu : float, optional (default = 0)
+        The expected value (mean) of the underlying normal distribution.
+        Acts as the location parameter of the distribution.
+    sigma : float, optional (default = 1)
+        The standard deviation of the underlying normal distribution.
+        Also acts as the scale parameter of distribution.
+
+    Examples
+    --------
+    >>> import numpy
+    >>> import paramnormal as pn
+    >>> numpy.random.seed(0)
+    >>> pn.truncated_normal(lower=-0.5, upper=0.5).rvs(size=3)
+    array([ 0.04687082,  0.20804061,  0.09879796])
+
+    >>> # you can also use greek letters
+    >>> numpy.random.seed(0)
+    >>> pn.truncated_normal(lower=-0.5, upper=2.5, σ=2).rvs(size=3)
+    array([ 0.8902748 ,  1.37377049,  1.04012565])
+
+    >>> # silly fake data
+    >>> numpy.random.seed(0)
+    >>> data = pn.truncated_normal(lower=-0.5, upper=2.5, mu=0, sigma=2).rvs(size=37)
+    >>> # pretend `data` is unknown and we want to fit a dist. to it
+    >>> pn.truncated_normal.fit(data)
+    params(lower=1.040124, upper=1.082447, mu=-8.097877e-06, sigma=1.033405)
+
+    In scipy, the distribution is defined as
+    ``stats.truncnorm(a, b, loc, scale)`` where
+
+    .. math::
+
+        a = \frac{\mathrm{lower bound}} - \mu}{\sigma}
+
+    and
+
+    .. math::
+
+        b = \frac{x_{\mathrm{upper bound}} - \mu}{\sigma}
+
+    Since ``a`` and ``b`` are directly linked to the location and scale
+    of the distribution as well as the lower and upper limits,
+    respectively, it's difficult to use the ``fit`` method of this
+    distirbution without either knowing a lot about it `a priori` or
+    assuming just as much.
+
+    References
+    ----------
+    http://scipy.github.io/devdocs/generated/scipy.stats.truncnorm
+    https://en.wikipedia.org/wiki/Rice_distribution
```

Comment:
Good catch and I didn't know about `fit_loc_scale`! Will check it out.

On Fri, Jul 29, 2016 at 5:03 AM Evgeni Burovski <notifications@github.com>
wrote:

> In paramnormal/dist.py
> <https://github.com/phobson/paramnormal/pull/36#discussion_r72780884>:
>
> > +    and
> > +
> > +    .. math::
> > +
> > +        b = \frac{x_{\mathrm{upper bound}} - \mu}{\sigma}
> > +
> > +    Since ``a`` and ``b`` are directly linked to the location and scale
> > +    of the distribution as well as the lower and upper limits,
> > +    respectively, it's difficult to use the ``fit`` method of this
> > +    distirbution without either knowing a lot about it `a priori` or
> > +    assuming just as much.
> > +
> > +    References
> > +    ----------
> > +    http://scipy.github.io/devdocs/generated/scipy.stats.truncnorm
> > +    https://en.wikipedia.org/wiki/Rice_distribution
>
> Rice?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phobson/paramnormal/pull/36/files/c30f4bcfcc2a0e8b426c322d7eb383556547d7af#r72780884>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABHCowITbToKZ4dE3_fR0uKTVadumbB6ks5qaev4gaJpZM4HvL4U>
> .
>